AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -176,7 +178,14 @@ - (void)_populateSupplementaryNodesOfKind:(NSString *)kind withMutableNodes:(NSM
     for (NSUInteger j = 0; j < rowCount; j++) {
       NSIndexPath *indexPath = [sectionIndexPath indexPathByAddingIndex:j];
       [indexPaths addObject:indexPath];
-      [nodes addObject:[self.collectionDataSource dataController:self supplementaryNodeOfKind:kind atIndexPath:indexPath]];
+      ASCellNodeBlock supplementaryCellBlock;
+      if (_dataSourceImplementsSupplementaryNodeBlockOfKindAtIndexPath) {
+        supplementaryCellBlock = [self.collectionDataSource dataController:self supplementaryNodeBlockOfKind:kind atIndexPath:indexPath];
```

Comment:
Hmm, I wonder if we need to set the same hierarchy state and layout delegate thing here?  I bet we are not actually pre-displaying or maybe not even calling fetchData on supplementary nodes...this goes back to my concern about how they are integrated, and the need to support headers and footers in tables.  cc @levi, @rahul-malik